NitroxClient/GameLogic/Fires.cs

```diff
@@ -0,0 +1,149 @@
+        /// </summary>
+        /// <param name="fireGuid">Guid of the Fire. Used for identification when dousing the fire</param>
+        /// <param name="subRootGuid">Guid of the Cyclops <see cref="SubRoot"/></param>
+        /// <param name="room">The room the Fire will be spawned in</param>
+        /// <param name="spawnNodeIndex">Each <see cref="CyclopsRooms"/> has multiple static Fire spawn points called spawnNodes. If the wrong index is provided,
+        ///     the clients will see fires in different places from the owner</param>
+        public void Create(string fireGuid, Optional<string> subRootGuid, CyclopsRooms room, int spawnNodeIndex)
+        {
+            // A Fire is either on a Cyclops or not. Currently, creating a Fire is only possible if it's in a Cyclops.
+            // I have not found any other code that creates a fire
+            if (subRootGuid.IsPresent())
+            {
+                SubFire subFire = GuidHelper.RequireObjectFrom(subRootGuid.Get()).GetComponent<SubRoot>().damageManager.subFire;
+                Dictionary<CyclopsRooms, SubFire.RoomFire> roomFiresDict = (Dictionary<CyclopsRooms, SubFire.RoomFire>)subFire.ReflectionGet("roomFires");
+                // Copied from SubFire_CreateFire_Patch, which copies from SubFire.CreateFire()
+                Transform transform2 = roomFiresDict[room].spawnNodes[spawnNodeIndex];
+
+                // If a fire already exists at the node, replace the old Guid with the new one
+                if (transform2.childCount > 0)
+                {
+                    Fire existingFire = transform2.GetComponentInChildren<Fire>();
+
+                    if (GuidHelper.GetGuid(existingFire.gameObject) != fireGuid)
+                    {
+                        Log.Error("[Fires.Create Fire already exists at node index " + spawnNodeIndex
+                            + "! Replacing existing Fire Guid " + GuidHelper.GetGuid(existingFire.gameObject)
+                            + " with Guid " + fireGuid
+                            + "]");
+
+                        GuidHelper.SetNewGuid(existingFire.gameObject, fireGuid);
+                    }
+
+                    return;
+                }
+
+                List<Transform> availableNodes = (List<Transform>)subFire.ReflectionGet("availableNodes");
+                availableNodes.Clear();
+                foreach (Transform transform in roomFiresDict[room].spawnNodes)
+                {
+                    if (transform.childCount == 0)
+                    {
+                        availableNodes.Add(transform);
+                    }
+                }
+                roomFiresDict[room].fireValue++;
+                PrefabSpawn component = transform2.GetComponent<PrefabSpawn>();
+                if (component == null)
+                {
+                    return;
```

Comment:
Should we log an error?

Reply:
Added an error to log if it didn't get a PrefabSpawn